tests/unit/test_gitlab_http_methods.py

```diff
@@ -38,6 +38,100 @@ def resp_cont(url, request):
             gl.http_request("get", "/not_there")
 
 
+def test_http_request_302_get(gl):
+    """Test to show that a redirect of a GET will not cause an error"""
+
+    @urlmatch(
+        scheme="http", netloc="localhost", path="/api/v4/user/status", method="get"
+    )
+    def resp_cont(url, request):
+        req = requests.Request(
+            "GET",
+            url="http://localhost//api/v4/user/status",
+        )
+        prepped = req.prepare()
+
+        history = [
+            response(
+                status_code=302,
+                content=(
+                    b"<html>\r\n<head><title>302 Found</title></head>\r\n<body bgcolor"
+                    b'="white">\r\n<center><h1>302 Found</h1></center>\r\n<hr><center>'
+                    b"nginx</center>\r\n</body>\r\n</html>\r\n"
+                ),
+                headers={"Location": "https://example.com/api/v4/user/status"},
+                reason="Moved Temporarily",
+                request=prepped,
+            )
+        ]
+
+        content = (
+            b'{"emoji":"thumbsup","message":"Hi redirect!","availability":"not_set",'
+            b'"message_html":"Hi redirect!","clear_status_at":null}'
+        )
+        resp_obj = response(
+            status_code=200,
+            content=content,
+            headers={},
+            reason="OK",
+            elapsed=5,
+            request=request,
+        )
+        resp_obj.history = history
+        return resp_obj
+
+    with HTTMock(resp_cont):
+        gl.http_request("get", "/user/status")
+
+
+def test_http_request_302_put(gl):
+    """Test to show that a redirect of a PUT will cause an error"""
+
+    @urlmatch(
+        scheme="http", netloc="localhost", path="/api/v4/user/status", method="put"
+    )
+    def resp_cont(url, request):
+        req = requests.Request(
+            "PUT",
+            url="http://localhost//api/v4/user/status",
+            data={"message": "Hi there!", "emoji": "thumbsup"},
+        )
```

Comment:
req is not used anymore, so maybe we can save a line?

```suggestion
        ).prepare()
```

Reply:
Personally I like the explicitness of it as I think it helps readability. But I can change if desired.

Reply:
Will mark as resolved. Let me know if still have questions/issues.

Reply:
How about we resolve this in a follow-up issue by removing `httmock` entirely in favor of `responses` where this can be rewritten potentially in a few lines with `rsps.add` :)

Reply:
Sounds reasonable to me to resolve this in a follow-up PR. I will get a chance to learn about `responses` 😊

Reply:
I opened a follow-up issue for this: https://github.com/python-gitlab/python-gitlab/issues/1591